### PR TITLE
[DOCS] Adds conceptual docs about multiple NLP model deployments

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR:
* adds a paragraph about the possibility to deploy the same model multiple times,
* reviews the paragraphs about `number of allocations` and `thread per allocation` setting and makes them more succinct,
* breaks the page sections into sub-pages.

### Preview

[Deploy the model in your cluster](https://stack-docs_2400.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-deploy-model.html)

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 1]
```diff
@@ -115,36 +115,48 @@ eland_import_hub_model --url https://<hostname>:<port> --es-api-key <api-key> ..
...
-{ref}/start-trained-model-deployment.html[start trained model deployment API] or
-specify the `--start` option when you run the `eland_import_hub_model` script.
+{ref}/start-trained-model-deployment.html[start trained model deployment API].
+
+You can deploy a model multiple times by assigning a unique deployment ID to 
+each deployment. It enables you to have dedicated deployments for different 
+purposes, such as search and ingest. By doing so, you ensure that the search 
+speed remains unaffected by ingest volume, and vice versa. Having separate 
+deployments for search and ingest mitigates performance issues resulting from 
+interactions between the two, which can be hard to diagnose.
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:132]:
```suggestion
You can deploy a model multiple times by assigning a unique deployment ID when starting
the deployment. It enables you to have dedicated deployments for different 
purposes, such as search and ingest. By doing so, you ensure that the search 
speed remains unaffected by ingest workloads, and vice versa. Having separate 
deployments for search and ingest mitigates performance issues resulting from 
interactions between the two, which can be hard to diagnose.
```

Comment (davidkyle) [reply docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc]: Plus can we add something about how the queue works

## Inference Request Queues
Every allocation of a model deployment has a dedicated queue to buffer inference requests. The size of this queue is controlled by the queue_capacity parameter to the start model deployment API. Once the queue is full any new requests will be rejected until some of the queued requests have been processed and there is capacity once again. Where multiple ingest pipelines reference the same model deployment it is possible for the queue to fill up resulting in rejected requests. Consider using dedicated deployments in this situation. 
 
Inference requests originating from search such as the text_expansion query or the query_vector_builder option in knn search are prioritised over non search requests.  The inference ingest processor creates normal priority requests, if a search query and ingest processor are using the same model deployment the higher priority search requests skip to the front of the queue to be processed before the lower priority ingest requests. This makes search responses lively while slowing down ingest where response time is not critical.

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 2]
```diff
@@ -115,36 +115,48 @@ eland_import_hub_model --url https://<hostname>:<port> --es-api-key <api-key> ..
...
+separate deployments for search and ingest mitigates performance issues 
+resulting from interactions between the two, which can be hard to diagnose.
+
+[role="screenshot"]
+image::images/ml-nlp-deployment-id.png["Model deployment on the Trained Models UI."]
+
+It is recommended to fine-tune each deployment based on its specific purpose. To 
+improve ingest performance, increase throughput by adding more allocations to 
+the deployment. For improved search speed, increase the number of threads per 
+allocation.
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:140]: Can we add a section on the deployment queues

## Inference request queues

Each allocation of a model deployment has a dedicated queue to buffer inference requests. The size of this queue is controlled by the queue_capacity parameter on the start model deployment API. Once the queue is full any new requests will be rejected until a queued request has been processed freeing capacity in the queue. If multiple ingest pipelines are using the deployment it is possible the queue will fill in which case consider using multiple deployments, increasing the queue capacity or when using bulk upload send a small number of documents at a time.

Inference requests originating from search such as the text_expansion query or the query_vector_builder option in knn search have a higher priority and skip to the front of the queue to be processed before any lower priority requests. The inference ingest processor creates normal priority requests, if the queue contains inference requests from ingest the search requests will be processed first, this makes search response lively at the expense of slowing down ingest which is not a time critical operation. This behaviour is designed to give the best performance where a single deployment is used for both search and ingest as an alternative to creating a dedicated deployment for each.

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 3]
```diff
@@ -157,8 +169,29 @@ change the number of allocations, you can use the
 {ref}/update-trained-model-deployment.html[update trained model stats API] after
 the allocation status is `started`.
 
-
 [discrete]
+[[infer-request-queues]]
+=== {infer-cap} request queues
+
+Each allocation of a model deployment has a dedicated queue to buffer {infer} 
+requests. The size of this queue is determined by the `queue_capacity parameter`
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:177]:
```suggestion
requests. The size of this queue is determined by the `queue_capacity` parameter
```

**Comment by szabosteve** [comment]
@elasticmachine test this please

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 4]
```diff
@@ -157,8 +169,29 @@ change the number of allocations, you can use the
...
+requests. The size of this queue is determined by the `queue_capacity parameter` 
+in the 
+{ref}/start-trained-model-deployment.html[start trained model deployment API]. 
+When the queue reaches its maximum capacity, new requests are declined until 
+some of the queued requests are processed, creating available capacity once 
+again. When multiple ingest pipelines reference the same deployment, the queue 
+can fill up, resulting in rejected requests. Consider using dedicated 
+deployments to prevent this situation.
+
+{infer-cap} requests originating from search, such as the `text_expansion`
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:186]: Maybe link to the text_expansion docs

File: docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc [hunk 5]
```diff
@@ -157,8 +169,29 @@ change the number of allocations, you can use the
...
+in the 
+{ref}/start-trained-model-deployment.html[start trained model deployment API]. 
+When the queue reaches its maximum capacity, new requests are declined until 
+some of the queued requests are processed, creating available capacity once 
+again. When multiple ingest pipelines reference the same deployment, the queue 
+can fill up, resulting in rejected requests. Consider using dedicated 
+deployments to prevent this situation.
+
+{infer-cap} requests originating from search, such as the `text_expansion` 
+query, receive higher priority compared to non-search requests. The {infer}
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-deploy-models.asciidoc:187]:
```suggestion
query, receive a higher priority compared to non-search requests. The {infer} 
```

"receive higher priority" doesn't sound right. maybe plain "have a higher priority"

**Review by davidkyle — APPROVED** [review]
LTGM

Left a couple of minor edits

**Pull request merged**